examples/seq2seq/seq2seq_trainer.py

```diff
@@ -0,0 +1,179 @@
+            loss = loss.mean()  # mean() to average on multi-gpu parallel training
+
+        if self.args.gradient_accumulation_steps > 1:
+            loss = loss / self.args.gradient_accumulation_steps
+
+        if self.args.fp16 and _use_native_amp:
+            self.scaler.scale(loss).backward()
+        elif self.args.fp16 and _use_apex:
+            with amp.scale_loss(loss, self.optimizer) as scaled_loss:
+                scaled_loss.backward()
+        else:
+            loss.backward()
+
+        return loss.item()
+
+    def _compute_loss(self, logits, labels, ignore_index):
+        # assuming label_smoothing is in args
+        if self.args.label_smoothing == 0:
+            # Same behavior as modeling_bart.py
+            loss_fct = torch.nn.CrossEntropyLoss(ignore_index=ignore_index)
+            assert logits.shape[-1] == self.model.config.vocab_size
+            loss = loss_fct(logits.view(-1, logits.shape[-1]), labels.view(-1))
+        else:
+            lprobs = torch.nn.functional.log_softmax(logits, dim=-1)
+            loss, nll_loss = label_smoothed_nll_loss(
+                lprobs, labels, self.args.label_smoothing, ignore_index=ignore_index
+            )
+        return loss
+
+    def prediction_step(
+        self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor, Any]], prediction_loss_only: bool
+    ) -> Tuple[Optional[float], Optional[torch.Tensor], Optional[torch.Tensor]]:
+        """
+        Perform an evaluation step on :obj:`model` using obj:`inputs`.
+
+        Subclass and override to inject custom behavior.
+
+        Args:
+            model (:obj:`nn.Module`):
+                The model to evaluate.
+            inputs (:obj:`Dict[str, Union[torch.Tensor, Any]]`):
+                The inputs and targets of the model.
+
+                The dictionary will be unpacked before being fed to the model. Most models expect the targets under the
+                argument :obj:`labels`. Check your model's documentation for all accepted arguments.
+            prediction_loss_only (:obj:`bool`):
+                Whether or not to return the loss only.
+
+        Return:
+            Tuple[Optional[float], Optional[torch.Tensor], Optional[torch.Tensor]]:
+            A tuple with the loss, logits and labels (each being optional).
+        """
+        has_labels = any(inputs.get(k) is not None for k in ["labels", "lm_labels", "masked_lm_labels"])
+
+        inputs = self._prepare_inputs(inputs)
+
+        max_length = model.config.max_position_embeddings
+        with torch.no_grad():
+            if self.args.predict_from_generate:
+                logits_out = model.generate(
+                    inputs["input_ids"], attention_mask=inputs["attention_mask"], use_cache=True
+                )
+                # in case the batch is shorter than max length, the output should be padded
+                logits = self._pad_tensors_to_max_len(logits_out, max_length, model.config.pad_token_id)
```

Comment:
`Trainer` requires all `logits` and `labels` to be of same length, so padding to model max length

Reply:
can we relax that assumption?

Reply:
if `logits` and `labels` have uneven lengths across batches then this [line](https://github.com/huggingface/transformers/blob/master/src/transformers/trainer.py#L1261) breaks